client/components/ColorPicker.js

```diff
@@ -0,0 +1,66 @@
+export default class ColorPicker extends React.Component {
+
+    constructor(props) {
+        super(props);
+    }
+
+    generateColor() {
+        let convertRGBToHex = function(rgb) {
+            let hexRed = rgb.r.toString(16).toUpperCase();
+            if (hexRed.length < 2)
+                hexRed += hexRed;
+
+            let hexGreen = rgb.g.toString(16).toUpperCase();
+            if (hexGreen.length < 2)
+                hexGreen += hexGreen;
+
+            let hexBlue = rgb.b.toString(16).toUpperCase();
+            if (hexBlue.length < 2)
+                hexBlue += hexBlue;
+
+            return "#" + hexRed + hexGreen + hexBlue;
+        };
+
+        let generatePrimaryColor = function() {
+            // Ranges of bright colors
+            let ranges = [
+                [100, 255],
+                [50, 200],
+                [10, 100]
+            ];
+
+            // Select random range and remove
+            let r = ranges.splice(Math.floor(Math.random() * ranges.length), 1)[0];
+
+            // Pick a random number from within the range
+            return Math.floor(Math.random() * (r[1] - r[0])) + r[0];
```

Comment:
To make this code easier to read, can you do the following:

```
let [low, high] = r;
```

And then use `low` (resp. `high`) in place of `r[0]` (resp. `r[1]`).